python/lsst/meas/extensions/scarlet/source.py

```diff
@@ -116,6 +155,21 @@ def initSource(frame, peak, observation, bbox,
             # so skip this source
             return None
 
+    if hasEdgeFlux(source, edgeDistance):
+        # The detection algorithm implemented in meas_algorithms
+        # does not place sources within the edge mask
+        # (roughly 5 pixels from the edge). This results in poor
+        # deblending of the edge source, which for bright sources
+        # may ruin an entire blend. So we reinitialize edge sources
+        # to allow for shifting and return the result.
```

Comment:
Note that while shifting may be defined w/n scarlet, it does not seem to be defined in this package, which makes this comment less useful. Consider either a brief description here (and or somewhere else in this package) or a link to scarlet docs where shifting is talked about.